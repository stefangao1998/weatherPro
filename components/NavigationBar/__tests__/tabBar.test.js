import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {create} from 'react-test-renderer';

import Adapter from 'enzyme-adapter-react-16';
import { configure } from 'enzyme';

configure({adapter: new Adapter()});

import {TabBar} from '../TabBar';

describe('TabBar component', () => {
  it('should render without issues', () => {
    const component = shallow(<TabBar />, {context: ''});

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with props', () => {
    const component = shallow(
      <TabBar 
        title={'text'}
        showBackbutton
        backButtonColor={'black'}
        showText={'text'}
      />,
    );

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });
});
