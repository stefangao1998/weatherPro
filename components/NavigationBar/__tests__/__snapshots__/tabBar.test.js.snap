// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBar component should render with props 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 72,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
        "zIndex": 1,
      },
      Object {
        "backgroundColor": "white",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "position": "absolute",
        "top": 40,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
        }
      }
    >
      text
    </Text>
  </View>
  <ForwardRef
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "height": 36,
        "justifyContent": "center",
        "left": 16,
        "overflow": "hidden",
        "position": "absolute",
        "top": 32,
        "width": 36,
      }
    }
  >
    <Icon
      color="black"
      name="left"
      size={20}
    />
  </ForwardRef>
  <ForwardRef
    activeOpacity={0.7}
    style={
      Object {
        "lineHeight": 24,
        "position": "absolute",
        "right": 16,
        "top": 40,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#7284A5",
          "fontSize": 14,
          "lineHeight": 16,
        }
      }
    >
      text
    </Text>
  </ForwardRef>
</View>
`;

exports[`TabBar component should render without issues 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 72,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
        "zIndex": 1,
      },
      Object {
        "backgroundColor": "white",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "position": "absolute",
        "top": 40,
        "width": "100%",
      }
    }
  />
</View>
`;
